src/fromager/constraints.py

```diff
@@ -0,0 +1,54 @@
+import copy
+import logging
+import pathlib
+from io import TextIOWrapper
+
+from packaging.requirements import Requirement
+
+logger = logging.getLogger(__name__)
+
+
+class Constraints:
+    def __init__(self, data: dict[str, Requirement]):
+        self._data = data
+
+    def get_new_requirement(self, req: Requirement):
+        constraint = self._data.get(req.name)
+        new_req = copy.deepcopy(req)
+
+        if not constraint:
+            return new_req, None
+
+        # only allow "==" in constraints for now
+        if len(constraint.specifier) != 1 or not str(constraint.specifier).startswith(
+            "=="
+        ):
+            logger.debug(
+                f"{constraint} is not allowed. Only '<package>==<version>' constraints are allowed"
+            )
+            return new_req, None
+
+        new_req.specifier = constraint.specifier
+        return new_req, constraint
+
+
+def _parse(content: TextIOWrapper) -> Constraints:
```

Comment:
There's some logic in https://github.com/python-wheel-build/fromager/blob/main/src/fromager/commands/bootstrap.py#L13 that handles comments and blank lines. We can go ahead with the version you have here, but it would be good to refactor both functions to share a common base loop for dealing with lines within the file.

Reply:
Okay will do. I will make a separate patch for this